code/modules/mob/living/living.dm

```diff
@@ -884,6 +884,20 @@
 
 	apply_effect((amount*RAD_MOB_COEFFICIENT)/max(1, (radiation**2)*RAD_OVERDOSE_REDUCTION), EFFECT_IRRADIATE, blocked)
 
+/mob/living/anti_magic_check(magic = TRUE, holy = FALSE)
+	if(!magic && !holy)
+		return
+
+	if((magic && has_trait(TRAIT_ANTIMAGIC)) || (holy && has_trait(TRAIT_HOLY)))
+		return src
+
+	var/list/obj/item/item_list = list()
+	item_list |= get_equipped_items(TRUE)
+	item_list |= held_items
+	for(var/obj/O in item_list)
```

Comment:
uhm what the fuck is this? shouldn't you send the signal and just have everything that is antimagic be listening on the mob already?

Reply:
this seems super super wrong

Reply:
We haven't gotten around to having memory efficient ways of handling that, signals on anything other than the object itself are difficult to listen to.

The exchange here is less memory cost per antimagic component and less performance cost per equip/unequip in exchange for slightly poorer performance on this signal.

Reply:
what, how do datum components register for signals on their parent obj then that doesn't make any sense

Reply:
like they are getting signals sent to the atom and receiving them

Reply:
The component is a part of the item, not the mob

Reply:
why can't they register to listen for signals emitted to something that's not their parent

this isn't very flexible